libethash-cl/CLMiner.cpp

```diff
@@ -412,10 +417,12 @@ void CLMiner::workLoop()
             current = w;  // kernel now processing newest work
             current.startNonce = startNonce;
             // Increase start nonce for following kernel execution.
-            startNonce += results.hashCount * m_settings.localWorkSize;
-
+            startNonce += m_settings.globalWorkSize;
```

Comment:
Why this change?

Reply:
Because it is more efficient. Consider the 3 scenarios where we get here:
1 - All the work groups have terminated without finding a solution.
     In this case the number of hashes calculated will always be equal to the global work size.
2 - Some work groups have exited early because a new job has arrived.
     A new job has arrived. It doesn't matter what we increment to, the start hash will be reset
     for the new job anyway.
3 - Some work groups have exited early because a work group has found a solution.
     In this case we may increment past some nonces that were not calculated by the work
     groups that terminated early. This is no consequence since our chances of finding a solution
     depend on the # of nonces checked, that does not depend on the checked nonces being
     consecutive.